vignettes/finalsize.Rmd

```diff
@@ -0,0 +1,275 @@
+
+:::
+
+## Running `final_size`
+
+The final size of the epidemic in the population can then be calculated using the only function in the package, `final_size`. This example allows the function to fall back on the default options for the arguments `solver` (`"iterative"`) and `control` (an empty list).
+
+```{r}
+# calculate final size
+final_size_data <- final_size(
+  r0 = r0,
+  contact_matrix = contact_matrix,
+  demography_vector = demography_vector,
+  susceptibility = susc_uniform,
+  p_susceptibility = p_susc_uniform
+)
+
+# order demographic groups
+final_size_data$demo_grp <- factor(
+  final_size_data$demo_grp,
+  levels = demography_data$age.group
+)
+```
+
+### Visualise final sizes
+
+```{r fig.cap ="Final size of an SIR epidemic in each age group. The final size is the cumulative number of infections in each age group over the course of the epidemic, expressed as a proportion of the respective age group.", fig.width=5, fig.height=4, class.source = 'fold-hide'}
+# plot data
+ggplot(final_size_data) +
+  geom_col(
+    aes(
+      demo_grp, p_infected
+    ),
+    fill = "lightgrey",
+    col = "black"
+  ) +
+  scale_y_continuous(
+    labels = scales::percent,
+    limits = c(0, 1)
+  ) +
+  theme_classic() +
+  coord_cartesian(
+    expand = FALSE
+  ) +
+  labs(
+    x = "Age group",
+    y = "% Infected"
+  )
+```
+
+### Final size proportions to counts
+
+`finalsize` returns the _proportion_ of each age (and risk) group infected in an epidemic outbreak. The final _counts_ of individuals infected can be visualised as well, by multiplying the final proportion of each age group infected with the total number of individuals in that group.
+
+The example below show how this can be done.
+
+```{r}
+# prepare demography data
+demography_data <- contact_data$demography
+
+# merge final size counts with demography vector
+final_size_data <- merge.data.table(
+  final_size_data,
+  demography_data,
+  by.x = "demo_grp",
+  by.y = "age.group"
+)
+
+# reset age group order
+final_size_data$demo_grp <- factor(
+  final_size_data$demo_grp,
+  levels = contact_data$demography$age.group
+)
+
+# multiply counts with proportion infected
+final_size_data$n_infected <- final_size_data$p_infected *
+  final_size_data$population
+```
+
+```{r fig.cap="Final size of an epidemic outbreak in a population, for different values of infection $R_0$. Converting the final size proportions in each age group to counts shows that individuals aged 18 -- 65 make up the bulk of cases in most  outbreaks. This may be attributed to this being both the largest age range in the analysis (more years in this range than any other), and because more people fall into this wide range than others. Contrast this figure with the one above, in which similar _proportions_ of each age group are infected.", fig.width=5, fig.height=4, class.source = 'fold-hide'}
```

Comment:
```suggestion
```{r fig.cap="Final size of an epidemic outbreak in a population, for different values of infection $R_0$. Converting the final size proportions in each age group to counts shows that in this scenario, individuals aged 18 -- 65 make up the bulk of cases. This may be attributed to this being both the largest age range in the analysis (more years in this range than any other), and because more people fall into this wide range than others. Contrast this figure with the one above, in which similar _proportions_ of each age group are infected.", fig.width=5, fig.height=4, class.source = 'fold-hide'}
```

Reply:
Fixed offline